neuracore/core/data/synced_recording.py

```diff
@@ -362,8 +363,8 @@ def __getitem__(self, idx: Union[int, slice]) -> Union[SyncPoint, list[SyncPoint
         if idx < 0 or idx >= len(self):
             raise IndexError("Index out of range")
 
-        # Get sync point data
-        sync_point: SyncPoint = self._recording_synced.frames[idx]
+        # Get sync point data, use deepcopy to avoid the memory leak
```

Comment:
Please be a bit more thorough with comment. Imagine someone comes across this who was not in our discussion